dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4188,6 +4188,212 @@ class FunctionLength : public IFunction
+    FunctionRepeat() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionRepeat>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 2; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!arguments[0]->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        if (!arguments[1]->isInteger())
+            throw Exception(
+                fmt::format("Illegal type {} of second argument of function {}", arguments[1]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        if (executeRepeat<UInt8>(block, arguments, result)
+            || executeRepeat<UInt16>(block, arguments, result)
+            || executeRepeat<UInt32>(block, arguments, result)
+            || executeRepeat<UInt64>(block, arguments, result)
+            || executeRepeat<Int8>(block, arguments, result)
+            || executeRepeat<Int16>(block, arguments, result)
+            || executeRepeat<Int32>(block, arguments, result)
+            || executeRepeat<Int64>(block, arguments, result))
+        {
+            return;
+        }
+        else
+        {
+            throw Exception(fmt::format("Illegal argument of function {}", getName()), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+    }
+
+private:
+    template <typename IntType>
+    bool executeRepeat(
+        Block & block,
+        const ColumnNumbers & arguments,
+        const size_t result) const
+    {
+        const ColumnPtr column_string = block.getByPosition(arguments[0]).column;
+        const ColumnPtr column_repeat_times = block.getByPosition(arguments[1]).column;
+
+        auto col_res = ColumnString::create();
+        if (const auto * col_string = checkAndGetColumn<ColumnString>(column_string.get()))
+        {
+            // vector const
+            if (column_repeat_times->isColumnConst())
+            {
+                const ColumnConst * col_const_repeat_times = checkAndGetColumnConst<ColumnVector<IntType>>(column_repeat_times.get());
+                if (col_const_repeat_times == nullptr)
+                {
+                    return false;
+                }
+                auto repeat_times = col_const_repeat_times->getValue<IntType>();
+                vectorConst(col_string->getChars(),
+                            col_string->getOffsets(),
+                            accurate::lessOp(INT64_MAX, repeat_times) ? INT64_MAX : repeat_times,
```

Comment:
Why use `INT64_MAX`, looks like tidb use `INT32_MAX`. Check https://github.com/pingcap/tidb/blob/fb6f6a4fda0fa24ec1f5ba3f1d31225c7fc4814a/expression/builtin_string.go#L675.

Maybe better to keep it compatible.

Reply:
In Tiflash, how did we deal with `maxAllowedPacket`. I can not find similar code
`if uint64(byteLength)*uint64(num) > b.maxAllowedPacket {
		return "", true, handleAllowedPacketOverflowed(b.ctx, "repeat", b.maxAllowedPacket)
	}`

Reply:
I think no need to handle it for now. Please add a comment in this issue as a remainder. https://github.com/pingcap/tiflash/issues/3669

Reply:
Done. Also add `todo` in code.